app/src/main/java/com/example/dennis/leagueoflegions/Unit.java

```diff
@@ -99,4 +100,9 @@ public int getColor() {
     public void setColor(int color) {
         this.color = color;
     }
-}
+
+    public String getType()
+    {
+        return type;
+    }
```

Comment:
The polymorphism here is wrong, since you implemented getType in Base and Army too. Instead we should make Unit an abstract class and have public abstract String getType(); and then in Base have @Override public String getType() { return "base"; } and same with army